### PR TITLE
Loading icon in layertree close #820

### LINKED ISSUE #820: Loading icon
Related to: https://github.com/camptocamp/c2cgeoportal/issues/1708

### PR DESCRIPTION
**Pull request by olivierSemet**
Close #820 

live example [here](https://oliviersemet.github.io/ngeo/820-loading-icon-in-layertree/examples/contribs/gmf/apps/desktop/)

ping @fredj

### CODE REVIEW

**Comment by sbrunner** [comment]
The loading icon don't have the right size ...

File: src/services/decoratelayerloading.js [hunk 1]
```diff
@@ -14,35 +14,36 @@ goog.require('ngeo');
...
-  } else {
-    goog.asserts.assert(layer instanceof ol.layer.Layer);
-    sources = [layer.getSource()];
-  }
+ngeo.decorateLayerLoading = function(layer, $scope, opt_parentLayer) {
+
+  var source;
+
+  /**
+   * @type {Array<string>|null|string}
```
Comment (fredj) [inline src/services/decoratelayerloading.js]: `string` must be removed

File: src/services/decoratelayerloading.js [hunk 2]
```diff
@@ -14,35 +14,36 @@ goog.require('ngeo');
...
+   */
+  var incrementEvents = null;
+
+  /**
+   * @type {Array<string>|null|string}
```
Comment (fredj) [inline src/services/decoratelayerloading.js]: `string` must be removed

**Comment by olivierSemet** [comment]
@sbrunner  do you mention the shift effect when the icon change from load icon to state icon ?

**Comment by fredj** [comment]
The shift can be removed by adding the `fa-fw` class (Fixed Width Icons) ?

File: src/services/decoratelayerloading.js [hunk 3]
```diff
@@ -14,35 +14,36 @@ goog.require('ngeo');
  *
  *      <span ng-if="layer.loading">please wait</span>
  *
- * @typedef {function(ol.layer.Base, angular.Scope)}
+ * @typedef {function(ol.layer.Base, angular.Scope, ol.layer.Base)}
```
Comment (fredj) [inline src/services/decoratelayerloading.js]: should be `@typedef {function(ol.layer.Base, angular.Scope, ol.layer.Base=)}`

**Comment by olivierSemet** [comment]
@fredj @sbrunner 

- fa-fw added
- New version with decorateLayerLoading  service working for groups of groups
- Unitest for decorateLayerLoading

**Comment by sbrunner** [comment]
It's me or the loading arrow will not be stable?
Same for the print...
If yes => issue!

**Comment by olivierSemet** [comment]
@sbrunner what do you mean by "not be stable"?

**Comment by sbrunner** [comment]
The center moves then the arrows turn...

**Comment by sbrunner** [comment]
Tested with Firefox!

**Comment by sbrunner** [comment]
It's working with Chrome...

**Comment by olivierSemet** [comment]
@sbrunner  Tested with firefox 45.02, the center is good. Which version do you have? 
It seems to be a known issue from Font Awesome [here](https://github.com/FortAwesome/Font-Awesome/issues/671)

**Comment by sbrunner** [comment]
46 :-)

**Comment by olivierSemet** [comment]
@sbrunner tested on ff 46.0, it's working great on my side... Don't know what to do.

**Comment by fredj** [comment]
Let's merge this PR and fix the remaining issues later

**Comment by olivierSemet** [comment]
capture d ecran 2016-05-09 a 10 51 42

**Comment by fredj** [comment]
Thanks @olivierSemet

**Pull request merged**